pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +291,138 @@ def test_index_ctor_infer_periodindex(self):
+            idx = Index(vals, dtype=object)
             assert isinstance(idx, Index)
             assert idx.dtype == object
+        else:
+            idx = Index(vals)
+            assert isinstance(idx, DatetimeIndex)
 
-    def test_constructor_dtypes_datetime(self):
+    @pytest.mark.parametrize("cast_idx", [True, False])
+    @pytest.mark.parametrize("vals", [
+        np.array([np.timedelta64(1, 'D'), np.timedelta64(1, 'D')]),
+        [timedelta(1), timedelta(1)]
+    ])
+    def test_constructor_dyptes_to_timedelta(self, cast_idx, vals):
+        if cast_idx:
+            idx = Index(vals, dtype=object)
+            assert isinstance(idx, Index)
+            assert idx.dtype == object
+        else:
+            idx = Index(vals)
+            assert isinstance(idx, TimedeltaIndex)
 
-        for tz in [None, 'UTC', 'US/Eastern', 'Asia/Tokyo']:
-            idx = pd.date_range('2011-01-01', periods=5, tz=tz)
-            dtype = idx.dtype
+    @pytest.mark.parametrize("tz", [
+        None, 'UTC', 'US/Eastern', 'Asia/Tokyo'])
+    @pytest.mark.parametrize("values", [
+        # pass values without timezone, as DatetimeIndex localizes it
+        pd.date_range('2011-01-01', periods=5).values,
+        pd.date_range('2011-01-01', periods=5).asi8])
+    @pytest.mark.parametrize("klass", [pd.Index, pd.DatetimeIndex])
+    def test_constructor_dtypes_datetime(self, tz, values, klass):
+        idx = pd.date_range('2011-01-01', periods=5, tz=tz)
+        dtype = idx.dtype
 
-            # pass values without timezone, as DatetimeIndex localizes it
-            for values in [pd.date_range('2011-01-01', periods=5).values,
-                           pd.date_range('2011-01-01', periods=5).asi8]:
+        res = klass(values, tz=tz)
+        tm.assert_index_equal(res, idx)
 
-                for res in [pd.Index(values, tz=tz),
-                            pd.Index(values, dtype=dtype),
-                            pd.Index(list(values), tz=tz),
-                            pd.Index(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(values, dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-                # check compat with DatetimeIndex
-                for res in [pd.DatetimeIndex(values, tz=tz),
-                            pd.DatetimeIndex(values, dtype=dtype),
-                            pd.DatetimeIndex(list(values), tz=tz),
-                            pd.DatetimeIndex(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(list(values), tz=tz)
+        tm.assert_index_equal(res, idx)
 
-    def test_constructor_dtypes_timedelta(self):
+        res = klass(list(values), dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
+    @pytest.mark.parametrize("attr", ['values', 'asi8'])
+    @pytest.mark.parametrize("klass", [pd.Index, pd.TimedeltaIndex])
+    def test_constructor_dtypes_timedelta(self, attr, klass):
```

Comment:
move this to pandas/tests/indexes/timedeltas/test_construction (maybe duplicate)